pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1655,291 +1652,26 @@ func parseConfigItems(config *zconfig.EdgeDevConfig, ctx *getconfigContext) {
-			}
-
-		case "debug.default.loglevel":
-			newGlobalConfig.DefaultLogLevel = item.Value
-
-		case "debug.default.remote.loglevel":
-			newGlobalConfig.DefaultRemoteLogLevel = item.Value
-
-		case "storage.dom0.disk.minusage.percent":
-			i64, err := strconv.ParseUint(item.Value, 10, 32)
-			if err == nil {
-				newGlobalConfig.Dom0MinDiskUsagePercent = uint32(i64)
-			}
-			// Max diskspace for dom0 is 80%.
-			if newGlobalConfig.Dom0MinDiskUsagePercent > 80 {
-				err = fmt.Errorf("dom0MinDiskUsagePercent (%d) "+
-					"> maxAllowed (80). Setting it to 80.",
-					newGlobalConfig.Dom0MinDiskUsagePercent)
-				log.Errorf("parseConfigItems: %s", err)
-				newGlobalConfig.Dom0MinDiskUsagePercent = 80
-			}
-			log.Infof("Set storage.dom0MinDiskUsagePercent to %d",
-				newGlobalConfig.Dom0MinDiskUsagePercent)
-
-		case "storage.apps.ignore.disk.check":
-			newBool, err := strconv.ParseBool(item.Value)
-			if err != nil {
-				log.Errorf("parseConfigItems: bad bool value %s for %s: %s\n",
-					item.Value, key, err)
-				continue
-			}
-			newGlobalConfig.IgnoreDiskCheckForApps = newBool
-
-		default:
-			// Handle agentname items for loglevels
-			newString := item.Value
-			components := strings.Split(key, ".")
-			if len(components) == 3 && components[0] == "debug" &&
-				components[2] == "loglevel" {
-
-				agentName := components[1]
-				current := agentlog.LogLevel(gcPtr, agentName)
-				if current != newString && newString != "" {
-					log.Infof("parseConfigItems: %s change from %v to %v\n",
-						key, current, newString)
-					agentlog.SetLogLevel(&newGlobalConfig,
-						agentName, newString)
-				} else {
-					agentlog.SetLogLevel(&newGlobalConfig,
-						agentName, current)
-				}
-			} else if len(components) == 4 && components[0] == "debug" &&
-				components[2] == "remote" && components[3] == "loglevel" {
-				agentName := components[1]
-				current := agentlog.RemoteLogLevel(gcPtr, agentName)
-				if current != newString && newString != "" {
-					log.Infof("parseConfigItems: %s change from %v to %v\n",
-						key, current, newString)
-					agentlog.SetRemoteLogLevel(&newGlobalConfig,
-						agentName, newString)
-				} else {
-					agentlog.SetRemoteLogLevel(&newGlobalConfig,
-						agentName, current)
-				}
-			} else {
-				unknownConfigItem = true
-				err = fmt.Errorf("Unknown configItem %s value %s\n",
-					key, item.Value)
-				log.Errorf("parseConfigItems: %s", err)
-			}
-		}
-		if err != nil {
-			if unknownConfigItem {
-				newGlobalStatus.UnknownConfigItems[key] =
-					types.ConfigItemStatus{Value: item.Value, Err: err}
-			} else {
-				err = fmt.Errorf("bad value %s for %s. Error: %s\n",
-					item.Value, key, err)
-				newGlobalStatus.ConfigItems[key] =
-					types.ConfigItemStatus{Err: err}
-				log.Errorf("parseConfigItems: %s", err)
-			}
-			unknownConfigItem = false
-			err = nil
-		}
+		err := ctx.zedagentCtx.specMap.ParseItem(&ctx.zedagentCtx.globalConfig, item.Value, item.Key)
```

Comment:
Did u mean to use newGlobalStatus instead of &ctx.zedagentCtx.globalConfig here?

Reply:
newGlobalStatus is the status of the item right? not the actual config? For parseItems we require the config